### PR TITLE
tsification + small package upgrade (semver)

### PR DESCRIPTION
**Pull request by OfekShilon**

### CODE REVIEW

File: lib/compilation-env.ts [hunk 1]
```diff
@@ -41,12 +41,12 @@ import type {PropertyGetter} from './properties.interfaces.js';
...
 
-type PropFunc = (string, any?) => any;
+type PropFunc = (s: string, a?: any) => any;
 
 export class CompilationEnvironment {
     ceProps: PropertyGetter;
-    compilationQueue: CompilationQueue;
-    compilerProps: CompilerProps;
+    compilationQueue: CompilationQueue | undefined;
+    compilerProps: PropFunc;
```
Comment (OfekShilon) [inline lib/compilation-env.ts:49]: That was a real 'bug' (bad type)

File: lib/cache/on-disk.ts [hunk 2]
```diff
@@ -80,9 +82,14 @@ export class OnDiskCache extends BaseCache {
                 };
             })
             .filter(Boolean);
+
         // Sort oldest first
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore filter(Boolean) should have sufficed but doesn't
```
Comment (partouf) [inline lib/cache/on-disk.ts:88]: wouldn't casting/setting the type of info to CacheEntry[] fix this?

File: lib/compilers/d8.ts [hunk 3]
```diff
@@ -36,6 +36,7 @@ import {unwrap} from '../assert.js';
 import {BaseCompiler, SimpleOutputFilenameCompiler} from '../base-compiler.js';
 import {CompilationEnvironment} from '../compilation-env.js';
 import {logger} from '../logger.js';
+import '../global.js';
```
Comment (partouf) [inline lib/compilers/d8.ts:39]: what's this?

File: lib/cache/on-disk.ts (hunk 2 above)
Comment (OfekShilon) [reply lib/cache/on-disk.ts:88]: Afraid not, `sort` isn't a field of `CacheEntry`

Comment (partouf) [reply lib/cache/on-disk.ts:88]: it's part of arrays (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort)

**Pull request merged**